src/main/java/com/ibm/cloud/sdk/core/util/DateUtils.java

```diff
@@ -13,80 +13,224 @@
+  private static final String DIALOG_DATE_TIME                 = "yyyy-MM-dd HH:mm:ss";
+
+  // These formatters are used to format (serialize) date and date-time values.
+  // The use of ".withZone(UTC)" ensures that the output string will be
+  // the UTC representation of the date or date-time value.
+  private static final DateTimeFormatter rfc3339FullDateFormat =
+      DateTimeFormatter.ofPattern(RFC3339_FULL_DATE).withZone(ZoneOffset.UTC);      // "yyyy-MM-dd"
+  private static final DateTimeFormatter rfc3339DateTimeFormatter =
+      DateTimeFormatter.ofPattern(RFC3339_DATE_TIME_FMT).withZone(ZoneOffset.UTC);  // "yyyy-MM-ddTHH:mm:ss.SSSZ"
+
+
+  //
+  // These formatters are used for parsing date-time values.
+  //
+  private static final DateTimeFormatter rfc3339DateTimeParse =
+      DateTimeFormatter.ofPattern(RFC3339_DATE_TIME_PARSE);
+  private static final DateTimeFormatter rfc3339DateTimeParseNoColon =
+      DateTimeFormatter.ofPattern(RFC3339_DATE_TIME_PARSE_NOCOLON);
+  private static final DateTimeFormatter utcDateTimeWithoutTZ =
+      DateTimeFormatter.ofPattern(UTC_DATE_TIME_NO_TZ).withZone(ZoneOffset.UTC);
+  private static final DateTimeFormatter dialogDateTime =
+      DateTimeFormatter.ofPattern(DIALOG_DATE_TIME).withZone(ZoneOffset.UTC);
+  private static final DateTimeFormatter alchemyDateTime =
+      DateTimeFormatter.ofPattern(ALCHEMY_DATE_TIME).withZone(ZoneOffset.UTC);
+
+  private static final List<DateTimeFormatter> dateTimeParsers =
+      Arrays.asList(
+          rfc3339DateTimeParse,            // "yyyy-MM-dd'T'HH:mm:ss[.SSS]X",  optional ms, tz: 'Z' or -03:00
+          rfc3339DateTimeParseNoColon,     // "yyyy-MM-dd'T'HH:mm:ss[.SSS]XXX" optional ms, tz: 'Z' or -0300
+          utcDateTimeWithoutTZ,            // "yyyy-MM-dd'T'HH:mm:ss[.SSS]"    optional ms
+          dialogDateTime,                  // "yyyy-MM-dd HH:mm:ss"
+          alchemyDateTime                  // "yyyyMMdd'T'HHmmss"
+          );
+
+  private static Pattern isJustNumber = Pattern.compile("^\\d+$");
+
 
   // Hide the ctor since this is a utility class.
   private DateUtils() {
   }
 
   /**
-   * Formats the specified Date value as an OpenAPI "date"
-   * (a string of the form "yyyy-MM-dd").
+   * Formats the specified {@link Date} instance as an RFC 3339 "full-date" value (yyyy-MM-dd).<br>
+   * The {@link Date} instance is assumed to represent the start of the specified day in UTC time.
+   * No adjustment for timezone is performed while formatting the value.
    *
-   * @param d the Date instance to be formatted
-   * @return a string representing an OpenAPI "date" (yyyy-MM-dd).
+   * @param d the {@link Date} instance to be formatted
+   * @return a string containing the RFC3339 "full-date" representation of "d" (example: 2020-01-01).
+   * @throws DateTimeException if an error occurs during formatting
    */
   public static String formatAsDate(Date d) {
-    String s;
-    synchronized (rfc3339FullDateFormatter) {
```

Comment:
One thing to note about this class.  It was previously using the old SimpleDateFormat class which is not thread-safe.
The newer DateTimeFormatter class is thread-safe, so no need for synchronization now.